extension/src/runtime.ts

```diff
@@ -4,27 +4,30 @@ import * as path from "path";
 
 const xcrExtension = ".xcr";
 
-export const launchRuntime = (terminals: { [key: string]: vscode.Terminal }): void => {
+export const launchRuntime = (terminals: Map<string, vscode.Terminal>): void => {
     const rootPath = vscode.workspace.rootPath;
     const config = vscode.workspace.getConfiguration();
     const xcruntimePath = (fs.existsSync(config.xcruntime.path)) ? config.xcruntime.path : "xcruntime.exe";
     const isWindowsPlatform = /^win/.test(process.platform);
     const mono = (isWindowsPlatform) ? "" : "mono";
     const xcassemblies = `${rootPath}${path.sep}XCR${path.sep}xcassemblies`;
-    let xcrFileFound = false;
+    const xcrFiles = [];
+
     fs.readdirSync(xcassemblies).forEach(file => {
```

Comment:
Optional: you can simplify your code by using filter and map functions on your file list